File: src/client/lib/shared/models/task.dart
class Task {
  final int id;
  final int projectId;
  final int? developerId;
  final String title;
  final String description;
  final double hourlyRate;
  final double? hoursSpent;
  final String status; // todo, in_progress, submitted, paid
  final String? zipPath;

  Task({
    required this.id,
    required this.projectId,
    this.developerId,
    required this.title,
    required this.description,
    required this.hourlyRate,
    this.hoursSpent,
    required this.status,
    this.zipPath,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      projectId: json['project_id'],
      developerId: json['developer_id'],
      title: json['title'],
      description: json['description'],
      hourlyRate: (json['hourly_rate'] as num).toDouble(),
      hoursSpent: json['hours_spent'] != null ? (json['hours_spent'] as num).toDouble() : null,
      status: json['status'],
      zipPath: json['zip_path'],
    );
  }

  double get amountDue => hoursSpent != null ? hourlyRate * hoursSpent! : 0.0;
}

File: src/backend/app/models/task.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base

class Task(Base):
    __tablename__ = "tasks"

    id = Column(Integer, primary_key=True, index=True)
    project_id = Column(Integer, ForeignKey("projects.id"), nullable=False)
    developer_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    title = Column(String, index=True, nullable=False)
    description = Column(String, index=True)
    hourly_rate = Column(Float, nullable=False)
    hours_spent = Column(Float, nullable=True)
    status = Column(String, default="todo", index=True)  # todo | in_progress | submitted | paid
    zip_path = Column(String, nullable=True)

    project = relationship("Project", back_populates="tasks")
    developer = relationship("User", back_populates="tasks")